include/cooperative_perception/angle.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2022 Leidos
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef COOPERATIVE_PERCEPTION_ANGLE_HPP
+#define COOPERATIVE_PERCEPTION_ANGLE_HPP
+
+#include <complex>
+#include <units.h>
+#include <boost/math/constants/constants.hpp>
+#include "cooperative_perception/utils.hpp"
+
+namespace cooperative_perception
+{
+/**
+ * @brief Angle type for representing and manipulating angles
+ */
+class Angle
+{
+private:
+  std::complex<double> value_{};
+
+public:
+  Angle(units::angle::radian_t angle_val) noexcept : value_{ units::math::cos(angle_val), units::math::sin(angle_val) }
+  {
+  }
+
+  auto set_angle(units::angle::radian_t angle_val) noexcept -> void
+  {
+    value_.real(units::math::cos(angle_val));
+    value_.imag(units::math::sin(angle_val));
+  }
+
+  auto get_angle() const noexcept -> units::angle::radian_t
+  {
+    auto phase_angle = std::arg(value_);
+    if (phase_angle < 0.0)
+    {
+      phase_angle += 2.0 * boost::math::double_constants::pi;
+    }
+    return units::angle::radian_t{ phase_angle };
+  }
+
+  friend auto operator==(const Angle& lhs, const Angle& rhs) -> bool
+  {
+    return lhs.value_ == rhs.value_;
+  }
+};
+
+inline auto operator+=(Angle& lhs, const Angle& rhs) noexcept -> Angle&
+{
+  auto angle_sum = lhs.get_angle() + rhs.get_angle();
+  lhs.set_angle(angle_sum);
+  return lhs;
+}
+
+inline auto operator-=(Angle& lhs, const Angle& rhs) noexcept -> Angle&
+{
+  auto angle_sub = lhs.get_angle() - rhs.get_angle();
+  lhs.set_angle(angle_sub);
+  return lhs;
+}
+
+inline auto operator+(Angle lhs, const Angle& rhs) noexcept -> Angle
+{
+  lhs += rhs;
+  return lhs;
+}
+
+inline auto operator-(Angle lhs, const Angle& rhs) noexcept -> Angle
+{
+  lhs -= rhs;
+  return lhs;
+}
+
+inline auto operator*=(Angle& lhs, double rhs) noexcept -> Angle&
+{
+  lhs.set_angle(rhs * lhs.get_angle());
+  return lhs;
+}
+
+inline auto operator*(Angle lhs, double rhs) noexcept -> Angle
+{
+  lhs *= rhs;
+  return lhs;
+}
+
+inline auto operator*(double lhs, Angle rhs) noexcept -> Angle
+{
+  rhs *= lhs;
+  return rhs;
+}
+
+inline auto operator/=(Angle& lhs, double rhs) noexcept -> Angle&
+{
+  lhs.set_angle(lhs.get_angle() / rhs);
+  return lhs;
+}
+
+inline auto operator/(Angle lhs, double rhs) -> Angle
+{
+  lhs /= rhs;
+  return lhs;
+}
+
+namespace utils
+{
+auto almostEqual(const Angle& lhs, const Angle& rhs) -> bool
+{
+  return lhs.get_angle() == rhs.get_angle();
```

Comment:
This is not almost equal as described by the function name. It's an exact equality.

Reply:
The `get_angle()` function returns a `units::angle::radian_t`, and the `operator==` overload checks almost equality. It's listed as "won't fix" in the project's issues. We should put a comment above the operation explaining it. Maybe something like:

```cpp
// Despite its appearance, this operation checks for almost-equality. The get_angle() function returns a 
// units::angle::radian_t type. The operator==(...) overload checks for almost-equality. See
// https://github.com/nholthaus/units/issues/118 for a discussion.
```

Reply:
Hmm interesting. Definitely need a comment like that to explain.